File: kyulee/programmers-2/구명보트.java
import java.util.Arrays;

public class 구명보트 {
    public static int solution(int[] people, int limit) {
        Arrays.sort(people);
        int answer = 0;

        int left = 0;
        int right = people.length - 1;

        while (left <= right) {
            if (people[left] + people[right] <= limit) {
                left++;
            }
            right--;
            answer++;
        }

        return answer;
    }

    public static void main(String[] args) {
        int[] people = {70, 50, 80, 50};
        int limit = 100;

        int result = solution(people, limit);
        System.out.println(result);
    }
}

File: sanghoon/programmers-2/구명보트.py
from collections import deque
def solution(people, limit):
    answer = 0
    people = deque(sorted(people, reverse=True))
    print(people)

    while len(people) > 0 :
        if len(people) == 1 :
            answer += 1
            break
        if people[0] + people[-1] <= limit :
            answer += 1
            people.pop()
            people.popleft()
        else :
            answer += 1
            people.popleft()
    return answer

people = [70, 80, 50]
limit = 100
solution(people, limit)